Lot of needed maintaince changes

KOLANICH opened the pull request:
Renamed back to json_delta.
Replaced `ujson` with `mujson`, which automatically uses the best available JSON parsing lib.
Replaced `setup.py` with `setup.cfg` and added `pyproject.toml`.
Got rid of raw CLI scripts. Using entry points instead.
Added .gitignore.

KOLANICH committed:
Replaced `ujson` with `mujson`, which automatically uses the best available JSON parsing lib.
Replaced `setup.py` with `setup.cfg` and added `pyproject.toml`.
Got rid of raw CLI scripts. Using entry points instead.
Added .gitignore.

darthghandi commented:
Did you test the change with json_delta installed? https://pypi.org/project/json-delta/
I'm concerned the name change might conflict. Thanks for the PR!

KOLANICH committed:
Moved the metadata from `setup.cfg` into PEP 621-compliant `pyproject.toml`.

KOLANICH commented:
OK, I have renamed back. Also added a separate commit moving the metadata into `pyproject.toml` from `setup.cfg`. The versions of `setuptools` supporting PEP 621 have dropped old python versions. But it may be possible to build using the tools like `flit_core`, which had been supporting PEP 621 long before `setuptools` has it implemented.
